lib/src/leak_tracker.dart

```diff
@@ -24,11 +24,16 @@ ObjectTracker _theObjectTracker() {
 ///
 /// The leak tracking will function only for debug/profile/developer mode.
 /// See usage guidance at https://github.com/dart-lang/leak_tracker.
-void enableLeakTracking({LeakTrackingConfiguration? config}) {
+void enableLeakTracking({
+  LeakTrackingConfiguration? config,
+  bool resetIfEnabled = false,
+}) {
   assert(() {
     final theConfig = config ??= const LeakTrackingConfiguration();
-    if (_objectTracker.value != null)
-      throw StateError('Leak tracking is alredy enabled.');
+    if (_objectTracker.value != null) {
+      if (!resetIfEnabled) throw StateError('Leak tracking is alredy enabled.');
```

Comment:
Is there a reason why we allow both behaviours? Could we just always reset ?

Reply:
Normally leak tracking should be set once for application. So, the second one is a smell.

Reset is needed just for tests.